### PR TITLE
Chronos: solve overhead caused by context_manager

### PR DESCRIPTION
**Pull request by plusbang**
## Description

In https://github.com/intel-analytics/BigDL/pull/6886, we try to use `InferenceOptimizer.get_context()` to replace `torch._no_grad` and realize thread control during inference. This make the context manager inside the inference loop.
But during recent performance tests, I found that every time calling `predict`, `torch.set_num_threads()` is called with quite large time cost and inference latency becomes >20ms.
```python
for x, y in test_loader:
    st = time.time()
    yhat = forecaster.predict(x.numpy(), quantize=True)
    latency.append(time.time()-st)
```

Maybe it's not a good idea to explicitly use `InferenceOptimizer.get_context()` in `predict`. It's better to avoid calling `torch.set_num_threads()` over and over again during inference.

We workaround this issue by no using the context manger to reduce overhead. Later, we will try new method to use context manager more friendly with no overhead.

### 2. User API changes

None

### 3. Summary of the change 

1. remove `InferenceOptimizer.get_context()` to `torch._no_grad`
2. implement of thread control (mainly through two attr `thread_num` and `optimized_model_thread_num`)
**first**, during initialization of forecaster, set the two attr according to current thread num;
**then**, if users call `build_onnx/openvino/jit` or `quantize` or `optimize`, update `optimized_model_thread_num` according to input parameter `thread_num`;
**next**, when users call `predict` or `predict_with_onnx/openvino/jit`, check the two attr, modify `thread_num` to be same as `optimized_model_thread_num` and call `torch.set_num_threads`
(Notice: especially for `predict`, if uses choose `acceleration=False`, keep `thread_num`)

Therefore, `torch.set_num_threads` will be called at most one time during inference loop.

3. uts

### 4. How to test?
- [x] Unit test

### CODE REVIEW

**Comment by jason-dai** [comment]
Shouldn't we fix nano instead?

**Comment by plusbang** [comment]
> Shouldn't we fix nano instead?

The problem comes when we try to use `InferenceOptimizer.get_context()` to replace torch._no_grad and realize thread control during inference. According to performance test results, I found that every time calling `predict`, `torch.set_num_threads()` is called with quite large time cost and inference latency becomes >20ms. This is not a problem with nano and the usage in chronos need to fix instead.

**Comment by jason-dai** [comment]
> > Shouldn't we fix nano instead?
> 
> The problem comes when we try to use `InferenceOptimizer.get_context()` to replace torch._no_grad and realize thread control during inference. According to performance test results, I found that every time calling `predict`, `torch.set_num_threads()` is called with quite large time cost and inference latency becomes >20ms. This is not a problem with nano and the usage in chronos need to fix instead.

We should always use `InferenceOptimizer.get_context()`, and improve it as needed.

**Comment by TheaperDeng** [comment]
> > > Shouldn't we fix nano instead?
> > 
> > 
> > The problem comes when we try to use `InferenceOptimizer.get_context()` to replace torch._no_grad and realize thread control during inference. According to performance test results, I found that every time calling `predict`, `torch.set_num_threads()` is called with quite large time cost and inference latency becomes >20ms. This is not a problem with nano and the usage in chronos need to fix instead.
> 
> We should always use `InferenceOptimizer.get_context()`, and improve it as needed.

This is an API problem since `InferenceOptimizer.get_context()` does not have any issue.

In Chronos we prefer users to use `forecaster.predict` for inferencing so that we need to call the context manager inside `predict`. This make the context manager inside the inference loop, which cause some overhead.

Since we don't want users to call context manager explicitly in chronos, we temp workaround this issue by no using the context manger to reduce some overhead. In next PR, we will add a method `forecaster.eval()` to use context manager more friendly with no overhead.

**Review by TheaperDeng — COMMENTED** [review]
others LGTM

File: python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py [hunk 1]
```diff
@@ -1711,7 +1734,6 @@ def export_torchscript_file(self, dirname="fp32_torchscript",
         """
         from bigdl.nano.pytorch import InferenceOptimizer
         from bigdl.nano.utils.log4Error import invalidInputError
-        from .utils import get_exported_module
```
[the remark this answers is not in the document]
Comment (plusbang) [reply python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:1714]: > it's used in line 1756

In line18, we have `from bigdl.chronos.forecaster.utils import *`. And I add this method in util.py.
https://github.com/intel-analytics/BigDL/blob/e037f2ba6e8a1c69d74a25e3ecca7d19c21d044e/python/chronos/src/bigdl/chronos/forecaster/utils.py#L39
 So I think this line can be deleted.

[the remark this answers is not in the document]
Comment (TheaperDeng) [reply python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:1714]: yeah, make sense.

**Pull request merged**